### DISCUSSION TITLE
Allow users to specify their own portal element to use for withinPortal (application-wide)

### DISCUSSION THREAD

**Discussion by Nantris** [mantinedev/mantine#2218] [Feature requests]
The rationale for this is that we use multiple portals in our app and we need to show Mantine above one and below another. Specifically this issue is affecting tooltips in our app, causing them to appear above our portals (but we need `withinPortal` on the tooltips for them to show to begin with)

**Comment by rtivital** [comment]
You can add default props to Portal component on MantineProvider – https://mantine.dev/theming/default-props/
You can set `target` prop and all elements will use it.

**Reply by Nantris** [reply]
Thanks so much for your reply! 

That works great, only I'm realizing that we need to set some of these tooltips to use one portal, and the rest to use another. I suppose we're out of luck for now if we need that, but is there some hope that it might be supported in the future?

As it stands now, the only way all tooltips will show is it we set them all to our highest `zIndex`'d portal, but then they show over elements they shouldn't also. The alternative has some tooltips appearing underneath layers they're intended to appear above, in most cases completely blocked by those layers.

**Reply by Nantris** [reply]
@rtivital are the odds something like that would be supported in the future greater than zero? I assume it's unlikely, but there is a real use case - though it's probably rather edge case.

Unfortunately many tooltips won't work without `withinPortal`, but `withinPortal` breaks zIndex levels in our app without more granularity.

If we could leverage both our `zIndex: 2` portal and our `zIndex: 3` portal, then all the layering would work beautifully.

### Currently - tooltips appear above popover

![image](https://user-images.githubusercontent.com/6835891/188025429-78303224-0fc4-4fb6-a7ad-9aa42e846b20.png)

### Desired - tooltips appear beneath popover

![image](https://user-images.githubusercontent.com/6835891/188025446-c41b72c7-a686-42bc-a093-eb816dfe7486.png)

Unfortunately, despite the amazing flexibility of Mantine, this outcome is not currently achievable without choosing to break some other components' zIndex levels in a similar fashion.

**Reply by islami00** [reply]
Hi, have you had a look at using nested providers: https://mantine.dev/theming/mantine-provider/#nested-mantineproviders  to override the default target prop here?

That way you could have a provider that changes the target for the button's tooltip in this specific case.

**Reply by Nantris** [reply]
Wow that might be perfect - I'm going to look into that further! Thanks very much @islami00!
